discovery-provider/src/tasks/users.py

```diff
@@ -230,3 +236,13 @@ def get_metadata_overrides_from_ipfs(session, update_task, user_record):
         logger.warning(f'users.py | {user_metadata}')
 
     return user_metadata
+
+def user_replica_set_upgraded(user_record):
+    primary_replica_set_configured = user_record.primary is not None
+    if primary_replica_set_configured:
+        primary_replica_set_configured = user_record.primary > 0
+    secondary_replica_set_configured = user_record.secondaries is not None 
+    if secondary_replica_set_configured:
+        secondary_replica_set_configured = len(user_record.secondaries) > 0
+    upgraded_replica_set = primary_replica_set_configured or secondary_replica_set_configured
```

Comment:
dont we want `primary_replica_set_configured and secondary_replica_set_configured`? what if one upgraded but the other didnt?

Reply:
should we error? or handle this somehow

Reply:
i don't think so - how would that scenario occur? maybe take a look at how we interact with UserReplicaSetManager and the function signatures for some context